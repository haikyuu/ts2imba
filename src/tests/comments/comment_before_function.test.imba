import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
a()
/*
 * comment
 */
function x() {
  return;
}
'''

test 'comment_before_function' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()
