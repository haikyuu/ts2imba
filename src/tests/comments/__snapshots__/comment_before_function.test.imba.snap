// Vitest Snapshot v1

exports[`comment_before_function 1`] = `
"a()
###
# comment
###def x
	return
"
`;
